src/zenml/client.py

```diff
@@ -2275,6 +2276,228 @@ def delete_stack_component(
             component.name,
         )
 
+    def deploy_stack_component(
+        self,
+        ctx: click.Context,
+        name: str,
+        flavor: str,
+        cloud: str,
+        component_type: StackComponentType,
+        configuration: Optional[Dict[str, Any]] = None,
+        labels: Optional[Dict[str, Any]] = None,
+    ) -> Optional["ComponentResponseModel"]:
+        """Deploys a stack component.
+
+        Args:
+            ctx: The click context.
+            name: The name of the deployed stack component.
+            flavor: The flavor of the deployed stack component.
+            cloud: The cloud of the deployed stack component.
+            component_type: The type of the stack component to deploy.
+            configuration: The configuration of the deployed stack component.
+            labels: The labels of the deployed stack component.
+
+        Returns:
+            The deployed stack component.
+        """
+        from zenml.cli.stack_recipes import (
+            deploy,
+            get_outputs,
+        )
+
+        STACK_COMPONENT_RECIPE_DIR = "deployed_stack_components"
+
+        config = None
+        if configuration:
+            # convert configuration dict to a json object
+            config = json.dumps(configuration)
+
+        # set the stack component and flavor
+        component = {component_type.value: flavor}
+
+        # path should be fixed at a constant in the
+        # global config directory
+        path = Path(
+            os.path.join(
+                io_utils.get_global_config_directory(),
+                STACK_COMPONENT_RECIPE_DIR,
+            )
+        )
+
+        try:
+            # Create the new component model
+            ctx.invoke(
```

Comment:
Still think this a bit of an anti-pattern to invoke the CLI from within the client code. Could you extract this out somewhere else, so that the deployment is handled by some function that we invoke here (and that can then be used by the CLI deploy command as well as here in the client code?